py/dml/ctree.py

```diff
@@ -4166,7 +4167,7 @@ def read(self):
     def assign_to(self, dest, typ):
         '''output C statements to assign an lvalue'''
         typ = safe_realtype(typ)
-        if isinstance(typ, TExternStruct):
+        if isinstance(typ, StructType):
```

Comment:
I don't understand the implications of this change. Is it covered by a test?

Reply:
`DesignatedStructInitializer` was previously used only for extern structs, but I repurposed it for any struct type for the initializers codegen:d by the use of DML designated initializers.

Reply:
Does one of your tests cover this case?

Reply:
The designated initializer tests? I don't understand the question -- this initializer is used for designated initializers of all struct-like types except bitfields, so obviously it's covered by the test that tests out the functionality of designated initializers.

Reply:
OK, good. I didn't read the full context, wanted to know if this change somehow extends DML and thus needs new tests. I'll take your word for it.